sharing/resource_share.go

```diff
@@ -230,33 +197,155 @@ func ResourceShare() common.Resource {
+					rawMap := rawTfShare.(map[string]interface{})
+
+					var partitions []Partition
+					if rawMap["partition"] != nil {
+						rawPartitions := rawMap["partition"].(*schema.Set).List()
+						for _, plainPartitionRaw := range rawPartitions {
+							plainPartition := plainPartitionRaw.(map[string]interface{})
+							partitions = append(partitions, Partition{
+								Values: parsePartition(plainPartition),
+							})
+						}
+					}
+
+					tfObject := SharedDataObject{
+						Name:                     rawMap["name"].(string),
+						Comment:                  rawMap["comment"].(string),
+						DataObjectType:           rawMap["data_object_type"].(string),
+						SharedAs:                 rawMap["shared_as"].(string),
+						CDFEnabled:               rawMap["cdf_enabled"].(bool),
+						HistoryDataSharingStatus: rawMap["history_data_sharing_status"].(string),
+						Partitions:               partitions,
+					}
+					if existingShare.Name == tfObject.Name {
+						keepShare = true
+					}
+
+				}
+
+				if !keepShare {
+					apiChanges = append(apiChanges, ShareDataChange{
+						Action:     ShareRemove,
+						DataObject: existingShare,
+					})
+				}
+			}
+
+			for _, raw := range objectsToAdd {
+				rawMap := raw.(map[string]interface{})
+				var partitions []Partition
+				if rawMap["partition"] != nil {
+					rawPartitions := rawMap["partition"].(*schema.Set).List()
+					for _, plainPartitionRaw := range rawPartitions {
+						plainPartition := plainPartitionRaw.(map[string]interface{})
+						partitions = append(partitions, Partition{
+							Values: parsePartition(plainPartition),
+						})
+					}
+				}
+				updateObject := SharedDataObject{
+					Name:                     rawMap["name"].(string),
+					Comment:                  rawMap["comment"].(string),
+					DataObjectType:           rawMap["data_object_type"].(string),
+					SharedAs:                 rawMap["shared_as"].(string),
+					CDFEnabled:               rawMap["cdf_enabled"].(bool),
+					HistoryDataSharingStatus: rawMap["history_data_sharing_status"].(string),
+					Partitions:               partitions,
+				}
+
+				// Look at the api call back result and see if the share already exists
+				exists := false
+				for _, existingDataObject := range beforeSi.Objects {
+					// This is an exact match. What should happen when the match is partial
+					// updateObject doesn't have the SharedAs field, and thus Equal behaves differently
+					if existingDataObject.Equal(updateObject) {
+						exists = true
+						break
+					}
+				}
+
+				if !exists {
+					apiChanges = append(apiChanges, ShareDataChange{
+						Action:     ShareAdd,
+						DataObject: updateObject,
+					})
+				}
+			}
+
 			err = NewSharesAPI(ctx, c).update(d.Id(), ShareUpdates{
-				Updates: changes,
+				Updates: apiChanges,
 			})
 			if err != nil {
-				if d.HasChange("owner") {
-					// Rollback
-					old, new := d.GetChange("owner")
+				// Rollback
+				if d.HasChanges("owner", "comment") {
```

Comment:
```suggestion
				if d.HasChange("owner") {
```
Hold over from an older PR where I introduced the `Comment` field to the delta share object.